### PR TITLE
Stripped unused environment plugin from webpack config

### PR DESCRIPTION
**Pull request by jordanh**
While reviewing #453 I noticed that we're not actually using `webpack.EnvironmentPlugin` in practice.

Anything we want on the client is being distributed via the SSR and stuck in `window.__ACTION__`.

Anything we want on the server is being pulled from the environment directly.

This PR strips EnvironmentPlugin.

### CODE REVIEW

**Comment by mattkrick** [comment]
LGTM

File: yarn.lock [hunk 1]
```diff
@@ -6886,7 +6879,7 @@ webpack@2.1.0-beta.22:
     source-map "^0.5.3"
     supports-color "^3.1.0"
     tapable "~0.2.3"
-    uglify-js "~2.6.0"
+    uglify-js "~2.7.3"
```
Comment (mattkrick) [inline yarn.lock]: huh, seeing the lockfile diff is _very_ interesting!

**Comment by mattkrick** [comment]
```
[0] ERROR in ../~/cashay/lib/schema/cashay-loader.js!./server/utils/getCashaySchema.js
[0] Module build failed: Error: The module '/Users/mattkrick/code/action/node_modules/bcrypt/build/Release/bcrypt_lib.node'
[0] was compiled against a different Node.js version using
[0] NODE_MODULE_VERSION 48. This version of Node.js requires
[0] NODE_MODULE_VERSION 51. Please try re-compiling or re-installing
[0] the module (for instance, using `npm rebuild` or`npm install`).
```

i can make this locally if i run node 7...same cashay loader that gives us crap

**Comment by mattkrick** [comment]
@jordanh hear me out... if our lockfile is generated on osx & then circleCI is running linux, then the binary it compiles isn't gonna agree with what's in the lockfile, right?

**Comment by mattkrick** [comment]
Ohhhh! we're looking for binaries in the post-test section. That could mean that the thing we build got GC'd or is somehow unavailable. Let's see if we can build in the same thread that we install in.

**Comment by jordanh** [comment]
I spent some real time digging through the CircleCI docs over the past week or so. This is important: https://circleci.com/docs/how-cache-works/

> Each branch of your project will have a separate cache. If it is the very first build for a branch, the cache from the default branch on GitHub or Bitbucket (normally master) will be used. If there is no cache for master, the cache from other branches will be used.

If the cache wasn't saved on `master` it will continue to fail here.

**Comment by jordanh** [comment]
Also: the `yarn.lock` file is system independent. It's only linking to specific module versions. Not binaries. So, it shouldn't matter that our build server is Linux and we are OSX.

**Pull request merged**